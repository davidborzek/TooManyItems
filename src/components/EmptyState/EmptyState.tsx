import { View, Text, StyleSheet } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { PropsWithChildren } from 'react';

type Props = {
  message: string;
  description?: string | null;
  icon?: React.ComponentProps<typeof Ionicons>['name'];
};

export default function EmptyState({
  message,
  description,
  icon,
  children,
}: PropsWithChildren<Props>) {
  return (
    <View style={styles.empty}>
      <Ionicons name={icon || 'grid-outline'} size={40} />
      <Text style={styles.message}>{message}</Text>
      {description && <Text style={styles.description}>{description}</Text>}
      {children}
    </View>
  );
}

const styles = StyleSheet.create({
  empty: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  message: {
    fontSize: 20,
    fontWeight: '600',
    marginVertical: 10,
  },
  description: {
    textAlign: 'center',
  },
});
