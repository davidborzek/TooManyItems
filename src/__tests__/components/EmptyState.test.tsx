import EmptyState from '../../components/EmptyState/EmptyState';
import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';

it('EmptyState renders correctly with required props', () => {
  const tree = renderer.create(<EmptyState message="Some message" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('EmptyState renders correctly with props', () => {
  const tree = renderer
    .create(
      <EmptyState
        message="Some message"
        description="Some description"
        icon="add"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('EmptyState renders correctly with children', () => {
  const tree = renderer
    .create(
      <EmptyState message="Some message">
        <Text>Custom children</Text>
      </EmptyState>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
