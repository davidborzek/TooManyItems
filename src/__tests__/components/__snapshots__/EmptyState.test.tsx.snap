// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState renders correctly with children 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text />
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "600",
        "marginVertical": 10,
      }
    }
  >
    Some message
  </Text>
  <Text>
    Custom children
  </Text>
</View>
`;

exports[`EmptyState renders correctly with props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text />
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "600",
        "marginVertical": 10,
      }
    }
  >
    Some message
  </Text>
  <Text
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Some description
  </Text>
</View>
`;

exports[`EmptyState renders correctly with required props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text />
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "600",
        "marginVertical": 10,
      }
    }
  >
    Some message
  </Text>
</View>
`;
